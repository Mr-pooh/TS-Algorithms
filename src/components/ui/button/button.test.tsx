import { render, screen } from "@testing-library/react";
import { Button } from "./button";

describe("Button", () => {
  it('render button', ()=> {
    render(<Button />);
    const btn = screen.getByRole('button');
    expect(btn).toMatchSnapshot();
  })
  
});
