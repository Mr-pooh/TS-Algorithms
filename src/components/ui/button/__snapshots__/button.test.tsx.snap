// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button render loader 1`] = `
<button
  class="text text_type_button text_color_primary button undefined loader "
  disabled=""
  type="button"
>
  <img
    alt="Загрузка."
    class="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button render with text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  >
    text
  </p>
</button>
`;

exports[`Button render without text 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;

exports[`Button rennder disable 1`] = `
<button
  class="text text_type_button text_color_primary button undefined false "
  disabled=""
  type="button"
>
  <p
    class="text undefined"
  />
</button>
`;
